### ISSUE TITLE
References don't work in HTML, if `\label` is not at the beginning of the line in Polytex

### ISSUE THREAD

**Issue by asmaier** [softcover/softcover#137]
Then converting from Polytex to HTML I noticed that references are not resolved in HTML, if the `\label` command is not at the beginning of the line.

E.g. the following example works 

    It also supports cross-referenced equations, as in Eq.~\eqref{eq:golden_ratio}.
    \begin{equation}
    \label{eq:golden_ratio}
    \phi = \frac{1+\sqrt{5}}{2} \approx 1.618 \qquad{\text{The Golden Ratio}}
    \end{equation}

But this example doesn't work (notice that we put the `\label` after the equation):

    It also supports cross-referenced equations, as in Eq.~\eqref{eq:golden_ratio}.
    \begin{equation}
    \phi = \frac{1+\sqrt{5}}{2} \approx 1.618 \qquad{\text{The Golden Ratio}} \label{eq:golden_ratio}
    \end{equation}

In the latter case the reference will not be resolved in the resulting HTML code.

**Comment by mhartl** [comment]
The LaTeX in question gets passed directly to [Mathjax](https://www.mathjax.org/), so handling any position-dependence is up to that project, not Softcover. In this case, if you put in an extra newline it should work:
```latex
It also supports cross-referenced equations, as in Eq.~\eqref{eq:golden_ratio}.
\begin{equation}
\phi = \frac{1+\sqrt{5}}{2} \approx 1.618 \qquad{\text{The Golden Ratio}} 
\label{eq:golden_ratio}
\end{equation}
```

**Comment by asmaier** [comment]
But Mathjax seems to handle the \label after the equation without problem. The following example works fine with pure MathJax:

    <!DOCTYPE html>
    <html>
    <head>
    <title>MathJax equation numbering</title>
    <meta http-equiv="Content-Type" content="text/html; charset=UTF-8" />
    <meta http-equiv="X-UA-Compatible" content="IE=edge" />

    <script src='https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML'>    </script>
    <script type="text/x-mathjax-config">
       MathJax.Hub.Config({ TeX: { equationNumbers: {autoNumber: "all"} } });
    </script>

    </head>
    <body>

    It also supports cross-referenced equations, as in Eq. ~\eqref{eq:golden_ratio3}.
    \begin{equation}
    \phi = \frac{1+\sqrt{5}}{2} \approx 1.618 \qquad{\text{The Golden Ratio}} \label{eq:golden_ratio3}
    \end{equation}

    </body>
    </html>

**Comment by mhartl** [comment]
Not sure what the source of the discrepancy is, but just put in a newline and it'll work.

**Comment by mhartl** [comment]
Wait, go back and read my [previous comment](https://github.com/softcover/softcover/issues/137#issuecomment-140207542). Softcover handles exactly the case you mention. Your original "broken" example is simply missing a newline.
